### DISCUSSION TITLE
Public Dashboards

### DISCUSSION THREAD

**Discussion by harshitgurani** [getredash/redash#6725] [General]
Hi , I have made a published dashboard public. My dashboard is working fine when I am using it but when I try to use it through the public links the widgets are loading continuously and No changes are happening for selected queries.  I checked if it could be issue of the sql query but for exact same query widgets 1 is working and another is not.

What can be the reason behind this?

**Comment by digitallawyer** [comment]
I am encountering the same or a similar issue with a public dashboard with filters:

- when applying changes to the non-public dashboard as a signed in user, date range and other filters apply as expected
- the public dashboard renders, but when applying a filter of any kind, the changes are not applied. Widgets show a loading indicator, but remain stuck in that state without ever updating.
- When forcing a reload of the public dashboard page (CTRL + R), filters are applied and charts are updated. 

Is this expected behavior. or are filters expected to update the current page on public dashboards without requiring a reload?

**Comment by ilpise** [comment]
I had same problem using redash 10.x.
Upgrading to redash 25.8.0 solved this.

With 25.8.0 another problem arises. With redash 10 I was able to run redash from a subfolder. This give me the opportunity to act to  public redash url (same domain) embedded iframe to set the iframe height, for example, using javascript.

In redash 25.8.0 the possibility to run redash in a subfolder seems not to be available. Then I have two different domains which are not compatible with the above javascript  'hack' to fit the iframe. At least the iframe height cannot be settled when and I have an additional vertical scrollbar related to the iframe.

With redash served from mydomain/redash/ the following code was working since the redash was proxied.
```
$("#redash_dashboard").append(ifrm)
// Adjusting the iframe height onload event
 ifrm.onload = function(){
     ifrm.style.height = ifrm.contentWindow.document.body.scrollHeight + 'px';
}
```

With redash 25.8.0 served from a different domain redash_mydomain the above script give me a 
```
Uncaught SecurityError: Failed to read a named property 'document' from 'Window': Blocked a frame with origin "https://mydomain.it" from accessing a cross-origin frame
```

Is there another way to embed redash public dashboards or a way to restore previous behavior with redash running in a subfolder?
